extensions/replication/queueProcessor/task.js

```diff
@@ -239,8 +239,28 @@ function initAndStart(zkClient) {
             }
         });
 
+        /**
+         * Get probe config will pull the configuration for the probe server based on
+         * the provided site key, and if the probe server config is not an array, it will
+         * return the global probe server config.
+         *
+         *  Get the probe server config for the first site in the site names list,
+         *  as if the probeConfig is an array there is only one element in site names
+         *
+         * @param {Object} queueProcessorConfig - Configuration of the queue processor that
+         *      holds the probe server configs for all sites
+         * @param {Array<String>} siteNames - List of site names
+         * @returns {ProbeServerConfig} Config for site or global config
+         */
+        function getProbeConfig(queueProcessorConfig, siteNames) {
+            if (Array.isArray(queueProcessorConfig.probeServer) && siteNames[0]) {
+                return queueProcessorConfig.probeServer.filter(probe => probe.site === siteNames[0])[0];
```

Comment:
this does not handle all cases.
* we may be started with an array of proble configs and a single siteName (s3c style) → ok
* we may be started with a single proble config and (one or more) site names (zenko style) → ok
* but we may also be started with an array of probe configs and multiple sites : in that case we should create multiple proble servers, one for each site... or at least generate an error, if this case is not supported

Reply:
> but we may also be started with an array of probe configs and multiple sites : in that case we should create multiple proble servers, one for each site... or at least generate an error, if this case is not supported

Normally this can't happen as Maha did a PR for that : https://github.com/scality/backbeat/pull/2537
Or am I missing something ?